util/run_manual_tests.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import re
+import subprocess
+from pathlib import Path
+
+import hjson
+import typer
+from pluralizer import Pluralizer
+from rich.console import Console
+from rich.markdown import Markdown
+from rich.prompt import Confirm
+from rich.text import Text
+
+pluralize = Pluralizer().pluralize
+
+
+def get_tty():
+    """Returns the terminal that we should use.
+
+    When this script is run directly, the terminal should be the terminal of the process.
+    When this script is run by bazel, the terminal should be the terminal of the bazel
+    client. Note that we cannot simply traverse the process tree, since this script is
+    run by the bazel server, which could be started from a different terminal.
+    """
+    term = subprocess.check_output(
+        ["/bin/ps", "-p", f"{os.getpid()}", "-o", "tty="]).decode().strip()
+    if term == "?":
+        res = subprocess.check_output(["/bin/ps", "-axo",
+                                       "tty=,cmd="]).decode()
+        m = re.search(r"^(?P<term>[\w/]*) *bazel.*test.*$", res, re.MULTILINE)
+        if not m:
+            raise RuntimeError("Could not find the terminal")
+        term = m["term"]
+    return f"/dev/{term}"
+
+
+# Set the terminal manually since this test is interactive.
+# This is really a hack since bazel does not support interactive commands.
+stdin = open(get_tty(), "r")
+stdout = open(get_tty(), "w")
+console = Console(force_interactive=True, force_terminal=True, file=stdout)
+
+
+def run_manual_testpoint(testplan_name, testpoint):
+    name = f"{testplan_name}/{testpoint['name']}"
+    md = f"**{name}**: {testpoint['desc']}"
+    console.print(Markdown(md))
+    console.print("")
+    prompt = Text.styled("\n> Was the test successful? [y/n]", style="bold")
+    return (name,
+            Confirm.ask(prompt,
+                        show_choices=False,
+                        console=console,
+                        stream=stdin))
+
+
+def print_results(name, results):
```

Comment:
nit: add docstring (or make private?) and type annotations?

Reply:
Done